MultiAuth: TypeError when session is initiated from a service

**Specifics of your environment**
1. Are you acting as SP/IdP/proxy? IdP
2. SimpleSAMLphp: What version are you using? 2.1.0 (I think all versions since 2.0 are affected)
3. PHP: What version are you using? 8.2
4. Platform: unix or Windows? unix
5. Webserver: Apache/Nginx/ISS? Apache

**Describe the bug**
When using simplesaml as single sign-on initiated from a service (to include `AuthnContextClassRef`) with multiple authsources through multiauth, after selecting an authsource an error appears (see below).

**To Reproduce**
Steps to reproduce the behavior:
1. Configure authsources and a service to use the SP (Test via Admin Interface does not trigger the bug)
2. Enable and configure multiauth and configure it like in the [documentation](https://simplesamlphp.org/docs/2.1/multiauth/multiauth.html) and set `'AuthnContextClassRef' => 'urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport',` for at least two sources
3. Start login at the service and select an authsource in the multiauth selection page
4. See error

**Expected behavior**
No error, like when using the selected authsource directly without MultiAuth or like when testing the multiauth through the admin interface.

**Screenshots or logs**
```
SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Backtrace:
2 src/SimpleSAML/Error/ExceptionHandler.php:35 (SimpleSAML\Error\ExceptionHandler::customExceptionHandler)
1 vendor/symfony/error-handler/ErrorHandler.php:541 (Symfony\Component\ErrorHandler\ErrorHandler::handleException)
0 [builtin] (N/A)
Caused by: TypeError: SimpleSAML\Module\multiauth\Auth\Source\MultiAuth::setPreviousSource(): Argument #1 ($source) must be of type string, int given, called in modules/multiauth/src/Controller/DiscoController.php on line 122
Backtrace:
6 modules/multiauth/src/Auth/Source/MultiAuth.php:237 (SimpleSAML\Module\multiauth\Auth\Source\MultiAuth::setPreviousSource)
5 modules/multiauth/src/Controller/DiscoController.php:122 (SimpleSAML\Module\multiauth\Controller\DiscoController::discovery)
4 vendor/symfony/http-kernel/HttpKernel.php:163 (Symfony\Component\HttpKernel\HttpKernel::handleRaw)
3 vendor/symfony/http-kernel/HttpKernel.php:75 (Symfony\Component\HttpKernel\HttpKernel::handle)
2 vendor/symfony/http-kernel/Kernel.php:202 (Symfony\Component\HttpKernel\Kernel::handle)
1 src/SimpleSAML/Module.php:234 (SimpleSAML\Module::process)
0 public/module.php:17 (N/A)
```

**Additional context**
The button ids and names on the selection page of multiauth are different (containing ints) when the login is initiated through a service. They are like `<input type="submit" name="sourceChoice[1]" class="pure-button" id="button-1" value="[...]">`. When testing through the admin interface the buttons have the correct ids like `<input type="submit" name="sourceChoice[auth-name]" class="pure-button" id="button-auth-name" value="[...]">`.


If you need more information please let me know.

--- a/modules/multiauth/src/Auth/Source/MultiAuth.php
+++ b/modules/multiauth/src/Auth/Source/MultiAuth.php
@@ -113,10 +113,10 @@ class MultiAuth extends Auth\Source
         ) {
             $refs = array_values($state['saml:RequestedAuthnContext']['AuthnContextClassRef']);
             $new_sources = [];
-            foreach ($this->sources as $source) {
+            foreach ($this->sources as $key => $source) {
                 $config_refs = $arrayUtils->arrayize($source['AuthnContextClassRef']);
                 if (count(array_intersect($config_refs, $refs)) >= 1) {
-                    $new_sources[] = $source;
+                    $new_sources[$key] = $source;
                 }
             }
             $state[self::SOURCESID] = $new_sources;

